docs: adjust remove_empty_groups in prometheus.exporter.process (#4979)
grafana/alloy: docs/sources/reference/components/prometheus/prometheus.exporter.process.md

| `track_children`      | `bool`   | Whether to track a process' children.             | `true`    | no       |
| `track_threads`       | `bool`   | Report metrics for a process' individual threads. | `true`    | no       |

If `remove_empty_groups` is `false` (the default), the process "groups" created by the `matcher` blocks continue to report metrics even after the processes in that group have stopped running. This ensures you can see when a process count drops to zero, but it can cause unbounded growth in reported metrics and memory usage if your `matcher` generates dynamic group names (e.g., using specific PIDs). The reporting will continue until {{< param "PRODUCT_NAME" >}} is restarted.

> When `remove_empty_groups` is set to `true`, process groups are forgotten and stop reporting metrics as soon as they contain no running processes. We recommend that you set `remove_empty_groups` to `true` if your name argument utilizes unique identifiers like `.PID` or `.StartTime`.

Reviewer: ```suggestion
When you set `remove_empty_groups` to `true`, process groups are forgotten and stop reporting metrics as soon as they contain no running processes.
Grafana recommends that you set `remove_empty_groups` to `true` if your name argument utilizes unique identifiers like `.PID` or `.StartTime`.
```
